src/DashboardUI/src/contexts/AppProvider.tsx

```diff
@@ -1,88 +1,93 @@
 import React from 'react';
-import { createContext, FC, useCallback, useContext, useEffect, useMemo, useRef, useState } from "react";
-import { useSearchParams } from "react-router-dom";
-import * as compress from "lz-string";
-import { IAuthenticationContext, useAuthenticationContext } from "../hooks/useAuthenticationContext";
+import { createContext, FC, useCallback, useContext, useEffect, useMemo, useRef, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
+import * as compress from 'lz-string';
+import { IAuthenticationContext, useAuthenticationContext } from '../hooks/useAuthenticationContext';
 import deepEqual from 'fast-deep-equal/es6';
-import { useDebounce } from "usehooks-ts";
+import { useDebounce } from 'usehooks-ts';
 
 interface AppContextState {
-  authenticationContext: IAuthenticationContext,
-  setUrlParam: (key: string, value: any) => void,
-  getUrlParam: <T, >(key: string) => T | undefined
+  authenticationContext: IAuthenticationContext;
+  setUrlParam: (key: string, value: any) => void;
+  getUrlParam: <T>(key: string) => T | undefined;
 }
 
 const defaultAppContextState = {
   authenticationContext: { isAuthenticated: false, user: null },
-  setUrlParam: (key: string, value: any) => { },
-  getUrlParam: <T,>(key: string): T | undefined => undefined
-}
+  setUrlParam: (key: string, value: any) => {},
+  getUrlParam: <T,>(key: string): T | undefined => undefined,
+};
 
 const AppContext = createContext<AppContextState>(defaultAppContextState);
-export const useAppContext = () => useContext(AppContext)
+export const useAppContext = () => useContext(AppContext);
 
 const AppProvider: FC = ({ children }) => {
-
   const [urlParams, setUrlParams] = useSearchParams();
 
   const authenticationContext = useAuthenticationContext();
 
-  const initUrlParams = useMemo(() => {
-    const stateStr = urlParams.get("state");
+  const initUrlParams = useMemo((): Record<string, any> => {
```

Comment:
We could always use `unknown` instead of `any` as long as the value won't change. This will satisfy the linter.

Reply:
True, true, we could do that. It doesn't really gain us anything though.

Reply:
Not really. It will just provide compilation errors if you try assigning the `unknown`. Gotta be casted like how objects work in .NET. I think it just depends if you want to enforce the linter `any` warning or not.

Reply:
Ah. Good point